flink-runtime/src/main/java/org/apache/flink/runtime/state/SharedStateRegistryImpl.java

```diff
@@ -365,4 +401,19 @@ public String toString() {
             return "EmptyDiscardStateObject{" + stateHandleID + '}';
         }
     }
+
+    private boolean preventsDiscardingCreatedCheckpoint(SharedStateEntry entry) {
+        // explicitly set by the backend, e.g. private state is reused
+        if (entry.preventDiscardingCreatedCheckpoint
+                && restoredCheckpointSharingStrategies.containsKey(entry.createdByCheckpointID)) {
+            return true;
+        }
+        // with NO_SHARING strategy, shared state, if any, is bundled inside the checkpoint folder
+        // noinspection RedundantIfStatement
```

Comment:
A typo?

Reply:
Hmm...I don't see it. What exactly do you mean?

Reply:
I'm confused about `noinspection RedundantIfStatement`, I guess you mean “don't check RedundantIfStatement”?

Reply:
Oh, I see. That was generated by the IDE in its format, so it understands that the warning about `if` should be suppressed
(I'd like to have an explicit `if` branch here).